src/storage/src/controller/hosts.rs

```diff
@@ -125,20 +123,17 @@ where
     }
 
     /// Provisions a storage host for the storage object with the specified ID.
-    ///
-    /// If `remote_addr` is `Some`, then the specified storage host is used.
-    /// Otherwise, a storage host is assigned automatically.
+    /// If the storage host is managed, this will ensure that the backing orchestrator
+    /// allocates resources; if not, it will ensure that the orchestrator will
+    /// release any resources. (For 'remote' storaged instances, the user is required
+    /// to independently make sure that any resources exist.)
```

Comment:
I think you could make more explicit that this method now provides upsert semantics.

Reply:
Should I specifically say "This may involve reconfiguring an existing service." like the orchestrator docs, or do you have something else in mind?